### PR TITLE
🏗 Allow array destructuring on preact hooks

### PR DESCRIPTION
**Pull request by jridgewell**

### CODE REVIEW

**Comment by jridgewell** [comment]
They're not approved by the linter, so it'll yell at you.

File: build-system/babel-plugins/babel-plugin-transform-simple-array-destructure/test/fixtures/transform/skip-element/output.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+{
+  const _temp = document,
+        second = _temp[1];
+  const _temp2 = document,
+        x = _temp2[0],
+        z = _temp2[2];
+}
```
Comment (kristoferbaxter) [inline build-system/babel-plugins/babel-plugin-transform-simple-array-destructure/test/fixtures/transform/skip-element/output.js]: Looks like terser undoes some of this 
```let[,n]=document,[o,,c]=document```

Comment (jridgewell) [reply build-system/babel-plugins/babel-plugin-transform-simple-array-destructure/test/fixtures/transform/skip-element/output.js]: Under what options?

File: build-system/babel-plugins/babel-plugin-transform-simple-array-destructure/test/fixtures/transform/deep-array/output.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const _temp = document,
+      _temp2 = _temp[0],
+      first = _temp2[0];
```
[the remark this answers is not in the document]
Comment (jridgewell) [reply build-system/babel-plugins/babel-plugin-transform-simple-array-destructure/test/fixtures/transform/deep-array/output.js]: Oh, that reminds me that I wanted to implement this with object destructure. Lol.

File: build-system/babel-plugins/babel-plugin-transform-simple-array-destructure/test/fixtures/transform/skip-element/output.js (hunk 1 above)
Comment (kristoferbaxter) [reply build-system/babel-plugins/babel-plugin-transform-simple-array-destructure/test/fixtures/transform/skip-element/output.js]: Pulled the options from our single pass configuration. But cannot reproduce now.

Looking into this again.

**Pull request merged**